modules/Makefile.kubernetes

```diff
@@ -0,0 +1,153 @@
+endif
+
+# Add the SSH endpoint to the command; everything after the host is expected to be a remote command
+KUBECTL_SSH_CMD += $(KUBECTL_SSH_TUNNEL)
+
+KUBECTL_CMD ?= $(KUBECTL_SSH_CMD) "$(KUBECTL)" --logtostderr=true --insecure-skip-tls-verify=true  
+
+# Set the namespace if one exists
+ifneq ($(KUBERNETES_NAMESPACE),)
+  KUBECTL_CMD += --namespace="$(KUBERNETES_NAMESPACE)" 
+endif
+
+# Set the URL to the kubernetes master, if one exists
+ifneq ($(KUBERNETES_URL),)
+  KUBECTL_CMD += --server="$(KUBERNETES_URL)" 
+endif
+
+# Set the user to authenticate against Kubernetes, if one exists
+ifneq ($(KUBERNETES_USER),)
+  KUBECTL_CMD += --username="$(KUBERNETES_USER)" 
+endif
+
+# Set the password to authenticate against Kubernetes, if one exists
+ifneq ($(KUBERNETES_PASS),)
+  KUBECTL_CMD += --password="$(KUBERNETES_PASS)" 
+endif
+
+DOCKERCFG_FILE ?= .dockercfg
+
+ifndef KUBERNETES_NAMESPACE
+  $(error KUBERNETES_NAMESPACE is undefined)
+endif
+
+export KUBERNETES_NAMESPACE
+
+#
+# Reference Docs:
+#  - https://cloud.google.com/container-engine/docs/kubectl/
+#
+## Display info about the kubernetes setup
+kubernetes\:info:
+	@echo -e "Cluster Namespace: $(call yellow,$(CLUSTER_NAMESPACE))"
+	@echo -e "Cluster Domain: $(call yellow,$(CLUSTER_DOMAIN))"
+	@echo -e "Kubernetes Namespace: $(call yellow,$(KUBERNETES_NAMESPACE))"
+	@echo -e "SSH Tunnel: $(call yellow,$(KUBECTL_SSH_TUNNEL))"
+	@echo -e "SSH User: $(call yellow,$(KUBECTL_SSH_USER))"
+
+## Create a new service
+kubernetes\:create-service:
+	$(call kubectl_create,service)
+
+## Create a new controller
+kubernetes\:create-controller:
+	$(call kubectl_create,controller)
+
+## Create a new app composed of a service and a controller
+kubernetes\:create:
+	@$(SELF) kubernetes:create-service kubernetes:create-controller
+
+## Replace an existing service with downtime
+kubernetes\:replace-service:
+	@echo -e "INFO: Replacing $(KUBERNETES_NAME) service on cluster $(call yellow,$(CLUSTER_NAMESPACE))..."
+	@$(SELF) kubernetes:delete-service kubernetes:create-service
+
+## Replace an existing controller with downtime
+kubernetes\:replace-controller:
+	@echo -e "INFO: Replacing $(KUBERNETES_NAME) controller on cluster $(call yellow,$(CLUSTER_NAMESPACE))..."
+	@$(SELF) kubernetes:delete-controller kubernetes:create-controller
+
+## Replace an app composed of a service and a controller with downtime
+kubernetes\:replace:
+	@$(SELF) kubernetes:replace-service kubernetes:replace-controller
+
+## Delete a new service
+kubernetes\:delete-service:
+	$(call kubectl_delete,service)
+
+## Delete a new controller
+kubernetes\:delete-controller:
+	$(call kubectl_delete,controller)
+
+## Delete a new app composed of a service and a controller
+kubernetes\:delete:
+	@$(SELF) kubernetes:delete-service kubernetes:delete-controller
+
+## Deploy controller and service; notify datadog
+kubernetes\:deploy:
```

Comment:
I propose we only expose this target. Its the only one anyone should ever call. Even the `replace` one is no longer used.

The fewer we expose, the clearer it will be